Make GRT a separate step

mleonowicz opened the pull request:
This PR introduces `grt` as a separate stage from `route`.

Now, stages may have multiple source stages instead of only one. This was needed, as the new `route` stage needs both `congestion.rpt` file, which comes from the `grt` route, and `4_cts.sdc` file, which is produced by a `cts` stage. 

Attaching `before` and `after` outputs produced by:
`bazel query "deps(//:L1MetadataArray_generate_abstract)" --output=graph --noimplicit_deps --notool_deps > graph && xdot graph`

![before](https://github.com/user-attachments/assets/b38905c5-28e0-4b36-a9f3-2f488a738a56)
![after](https://github.com/user-attachments/assets/1a73296c-88f5-4bc6-965e-ec86d632e852)

oharboe reviewed (changes requested):
- Set GENERATE_ARTIFACTS_ON_FAILURE=1 variable for all stages in ORFS. This will set ORFS into a mode where it creates an artifact for a failed operation, such as ORFS so that e.g. a global route failure artifacts can be downloaded and studied.
- Rebase on main

mleonowicz commented:
To support passing `GENERATE_ARTIFACTS_ON_FAILURE = 1` argument to all stages, I have added a new argument to `orfs_flow` called `common_args`.

I have also rebased the branch.
